src/aiida/orm/nodes/process/calculation/calcjob.py

```diff
@@ -135,6 +132,7 @@ def _hash_ignored_attributes(cls) -> Tuple[str, ...]:  # noqa: N805
             'priority',
             'max_wallclock_seconds',
             'max_memory_kb',
+            'version',  # ?? How I exclude the version.core ??
```

Comment:
@sphuber 
First, the attributes like `queue_name` is in `last_job_info` which is in `_node._updatable_attributes` and already be excluded, so the is no point to keep it there.
Second, it now only support exclude by key, and for the nest dictionary the login in `CalcJobNodeCaching` does not yet try to go level by level to exclude things. 

In order to make this work, I think in the `_hash_ignored_attributes` here it can specify `version.core` and in the NodeCaching class it will use the format to exclude precisely the version.core to have attribute object from: `{version: {'core': --, 'plugin': --}}` to `{'version': {'plugin': --}}`.

What you think?

Reply:
You _could_ change `_hash_ignored_attributes` and the implementation that consumes it to support nested keys, but you could also update `CalcJobNodeCaching._get_objects_to_hash()` to just call the `super` and then from the returned list, get the element that corresponds to the dictionary of attributes and then remove the 'version.core' key.

I am still not a 100% clear about the hypothetical case of the plugin being updated needing the cache to be invalidated (by the hash being different). Wouldn't the same reasoning also  not apply to `aiida-core`? What if the `CalcJob` base class changes some small (but significant) part of the implementation? Wouldn't that be the same scenario? So why keep the plugin version but drop the core version in that case? And if the changes could happen due to a fix of a bug (which could be released with a patch version of `aiida-core`) then just checking the major or even minor version _also_ won't protect against this hypothetical case.

So it seems we are back to square 1. Yes, keeping the core and plugin versions in the hash might be the safest, but how common are the cases that it is trying to protect against? If they are rare enough, should we really invalidate hashes with each (patch) version update, strongly reducing the effectiveness of the caching mechanism?